apps/mobile-wallet/src/features/address-deletion/useCanDeleteAddress.tsx

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AddressHash } from '@alephium/shared'
+
+import { useAppSelector } from '~/hooks/redux'
+import { selectAddressByHash } from '~/store/addressesSlice'
+
+const useCanDeleteAddress = (addressHash: AddressHash) => {
+  const address = useAppSelector((s) => selectAddressByHash(s, addressHash))
+
+  return address && !address.settings.isDefault && address.index !== 0
```

Comment:
Ah, I now get why I saw this "bug" of not being able to delete an address. It's because the address is at index 0. As a user this is a bit confusing IMO.

Reply:
I know, we should adapt our code to allow deleting index 0, but I am afraid it will require quite some changes so for now I think it's ok to ship the feature for all the rest of the addresses. WDYT?

Reply:
I remember we had a problem on DW when deleting the initial address (index 0) but tbh I can't think of why this would be a problem on mobile.

Reply:
Done